`tctl sso`: update general SSO docs and GitHub guide. (#14227)
gravitational/teleport: docs/pages/setup/admin/github-sso.mdx

(!docs/pages/includes/tctl.mdx!)

> ## Step 1/5. Create a GitHub OAuth app

Reviewer: Step 3/5 appears to be missing

[Creating an OAuth App](https://docs.github.com/en/developers/apps/building-oauth-apps/creating-an-oauth-app)

## Step 2/5. Define a GitHub authentication connector

Define a GitHub authentication connector by running the following command:

> ```code

Reviewer: In case a reader blindly copies and runs this command, I think we should include variables using the `${VARNAME?}` syntax. We can also include environment variable assignments for the client ID and client secret, plus a quick instruction to grab these from your newly created OAuth app

Author: Do we have an existing example of this approach in our docs?

Reviewer: Here's one example: https://github.com/gravitational/teleport/blame/348b290cc7db363ebb76f3553a4625b11fe60a1e/docs/pages/kubernetes-access/getting-started.mdx#L77-L89

        - access
```

> The values of `client_id` and `client_secret` come from the GitHub OAuth App you created earlier. `redirect_url` was automatically configured based on the settings of your Teleport cluster when you ran [`tctl sso configure github`](../reference/cli.mdx#tctl-sso-configure-github).

Reviewer: ```suggestion
The values of `client_id` and `client_secret` come from the GitHub OAuth App you created earlier. `redirect_url` was automatically configured based on the settings of your Teleport cluster when you ran `tctl sso configure github`.
```

<Admonition
  type="tip"
  title="Teams to logins mapping"
>

>   The `--teams-to-logins` flag (`-r` for short) can be repeated multiple times. Each repetition will create an additional entry in the `teams_to_logins` mapping of your GitHub authentication connector.

Reviewer: ```suggestion
When you run `tctl sso configure github`, you can include the `--teams-to-logins` flag (`-r` for short) multiple times. Each repetition will create an additional entry in the `teams_to_logins` mapping of your GitHub authentication connector.
```
To link this more explicitly to the `tctl sso configure` command.

  
  To map a GitHub team to multiple Teleport roles, list these roles after the name of the GitHub team, separated by commas: `-r org,team,role1,role2,role3,...`.
</Admonition>

Teleport will request only the `read:org` OAuth scope. Read more about OAuth scopes in [GitHub's documentation](https://developer.github.com/apps/building-oauth-apps/understanding-scopes-for-oauth-apps/).

> [GitHub OAuth scopes](https://developer.github.com/apps/building-oauth-apps/understanding-scopes-for-oauth-apps/)

Reviewer: ```suggestion
```

> To ensure the connector is working as intended: SSO flow is passing, roles are assigned as intended etc., you can now test it using [`tctl sso test`](../reference/cli.mdx#tctl-sso-test) command.

Reviewer: ```suggestion
To ensure that the SSO flow is working and roles are assigned as intended, you can now test your GitHub connector using the `tctl sso test` command.
```
Removing the link since we'll be explaining the command here. We can add links to the CLI reference in a "Next steps" section. Also tightening up this sentence a bit.

> 

Reviewer: I think we should add a sentence explaining what this command does.

```code
$ tctl sso test github.yaml
# If browser window does not open automatically, open it by clicking on the link:
#  http://127.0.0.1:62482/e1fadd3f-dc6a-420b-9791-94fd29dfc2db
# Success! Logged in as: ExampleGithubUser
# --------------------------------------------------------------------------------
# Authentication details:
#    logins:
#    - access
#    roles:
#    - access
#    traits:
#      github_teams:
#      - admins
#      kubernetes_groups: null
#      kubernetes_users: null
#      logins:
#      - ExampleGithubUser
#    username: ExampleGithubUser
# --------------------------------------------------------------------------------
# [GitHub] Received claims:
# organization_to_teams:
#   octocats:
#   - admins
# teams:
# - admins
# username: ExampleGithubUser
# 
# --------------------------------------------------------------------------------
# [GitHub] Connector team to logins mapping:
# - logins:
#   - access
#   organization: octocats
#   team: admins
# 
# --------------------------------------------------------------------------------
# For more details repeat the command with --debug flag.
```

> The test command can accept the auth connector from standard input. The pipeline may also utilise `tee` to save the connector to file:

Reviewer: ```suggestion
The test command can accept the auth connector from standard input. The pipeline may also utilize `tee` to save the connector to a file so you can define and test the connector in a single command:
```
I think the docs mostly use US spelling, so we can standardize on that. Also linked this sentence to the preceding text a bit more.

  able to determine team memberships for these organizations.
</Admonition>

## Step 4/5. Create the connector

> Once you are satisfied with the connector, you can finally create it using `tctl`:

Reviewer: ```suggestion
Once you are satisfied with the connector, you can create it using `tctl`:
```
